pkg/errors/codes/codes.go

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache v2.0 license.
+package codes
```

Comment:
Is there an advantage to putting codes in a separate package?

Reply:
I view codes as the most base level error abstraction. Then the errors pkg as our wrapper for codes to provide enhanced functionality that integrates with the LastError field. Seemed to make sense to me in terms of not crowding the errors pkg and separation of concerns, but happy to hear other thoughts on this.

Reply:
Closing to help me organize PR revisions, feel free to re-open if I didn't address something.